src/pytorch_lightning/loggers/wandb.py

```diff
@@ -251,17 +252,17 @@ def __init__(
         self,
         name: Optional[str] = None,
         save_dir: Optional[str] = None,
-        offline: Optional[bool] = False,
+        offline: bool = False,
         id: Optional[str] = None,
         anonymous: Optional[bool] = None,
         version: Optional[str] = None,
         project: Optional[str] = None,
         log_model: Union[str, bool] = False,
-        experiment=None,
-        prefix: Optional[str] = "",
+        experiment: Union[Run, RunDisabled, None] = None,
+        prefix: str = "",
         agg_key_funcs: Optional[Mapping[str, Callable[[Sequence[float]], float]]] = None,
         agg_default_func: Optional[Callable[[Sequence[float]], float]] = None,
-        **kwargs,
+        **kwargs: Any,
     ):
```

Comment:
nit: Shall we add `None` to the return type of `__init__`?

> Note that the return type of __init__ ought to be annotated with -> None. The reason for this is subtle. If __init__ assumed a return annotation of -> None, would that mean that an argument-less, un-annotated __init__ method should still be type-checked? Rather than leaving this ambiguous or introducing an exception to the exception, we simply say that __init__ ought to have a return annotation; the default behavior is thus the same as for other methods.
> https://peps.python.org/pep-0484/#the-meaning-of-annotations
```suggestion
    ) -> None:
```